include/experimental/__p1673_bits/blas1_linalg_swap.hpp

```diff
@@ -111,10 +122,10 @@ void swap_elements(
   std::experimental::mdspan<ElementType_x, std::experimental::extents<ext_x ...>, Layout_x, Accessor_x> x,
   std::experimental::mdspan<ElementType_y, std::experimental::extents<ext_y ...>, Layout_y, Accessor_y> y)
 {
-  if constexpr (x.rank() == 1) {
+  if constexpr (x.rank() == 1 && y.rank() == 1) {
     swap_rank_1(x, y);
   }
-  else if constexpr (x.rank() == 2) {
+  else if constexpr (x.rank() == 2 && y.rank() == 2) {
     swap_rank_2(x, y);
   }
```

Comment:
Did I write this `static_assert`?!?  It's definitely wrong.

1. `"Not implemented"` is `true` in a `bool` context, so `static_assert("Not implemented")` is just `static_assert(true)`.
2. One can't actually SFINAE out `static_assert(false)`.  Here is an example: https://godbolt.org/z/eY565W14n

One can google "dependent false" for a standard technique for dealing with (2).  However, we don't need it in this case; instead of an `else` clause, we could just have `else if constexpr (x.rank() > 2 or y.rank() > 2)`.

Reply:
I've filed this as a separate issue, https://github.com/kokkos/stdBLAS/issues/83 .